src/main/java/com/minecolonies/coremod/client/render/worldevent/ColonyWorldRenderMacros.java

```diff
@@ -0,0 +1,101 @@
+package com.minecolonies.coremod.client.render.worldevent;
+
+import com.ldtteam.structurize.util.WorldRenderMacros;
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.mojang.blaze3d.vertex.VertexConsumer;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.util.FastColor;
+import net.minecraft.world.phys.AABB;
+
+/**
+ * Extra {@link com.ldtteam.structurize.util.WorldRenderMacros}.  Maybe port it to Structurize at some point.
+ */
+public class ColonyWorldRenderMacros
+{
+    /**
+     * Render a wireframe box.
+     * @param poseStack         pose stack
+     * @param bufferSource      buffer source
+     * @param bounds            bounding box to draw
+     * @param width             line width
+     * @param color             line color (ARGB)
+     * @param showThroughBlocks true to render through existing blocks, false to only render in air
+     */
+    public static void renderLineBox(final PoseStack poseStack, final MultiBufferSource.BufferSource bufferSource,
+                                     final AABB bounds, final float width, final int color, final boolean showThroughBlocks)
+    {
+        final float halfLine = width / 2.0f;
+        final float minX = (float) (bounds.minX - halfLine);
+        final float minY = (float) (bounds.minY - halfLine);
+        final float minZ = (float) (bounds.minZ - halfLine);
+        final float minX2 = minX + width;
+        final float minY2 = minY + width;
+        final float minZ2 = minZ + width;
+
+        final float maxX = (float) (bounds.maxX + halfLine);
+        final float maxY = (float) (bounds.maxY + halfLine);
+        final float maxZ = (float) (bounds.maxZ + halfLine);
+        final float maxX2 = maxX - width;
+        final float maxY2 = maxY - width;
+        final float maxZ2 = maxZ - width;
+
+        final int red = FastColor.ARGB32.red(color);
+        final int green = FastColor.ARGB32.green(color);
+        final int blue = FastColor.ARGB32.blue(color);
+        final int alpha = FastColor.ARGB32.alpha(color);
+
+        renderLineBox(poseStack, bufferSource.getBuffer(RenderTypes.LINES_OUTSIDE_BLOCKS),
+                minX, minY, minZ, minX2, minY2, minZ2, maxX, maxY, maxZ, maxX2, maxY2, maxZ2,
+                red, green, blue, alpha);
+
+        if (showThroughBlocks)

```

Comment:
I would switch the order, look at hive image - the dull colour is above the full colour in visible part

Reply:
Switching `endBatch` didn't actually do anything.  Switching the draw order fixed it though.
![image](https://user-images.githubusercontent.com/539951/227035706-1efea250-d04b-4f4a-955a-8dad85144ce2.png)

Reply:
Very nice work!